modules/base/rendering/renderablenodeline.cpp

```diff
@@ -0,0 +1,307 @@
+
+    _vertexArray.push_back(_endPos.x);
+    _vertexArray.push_back(_endPos.y);
+    _vertexArray.push_back(_endPos.z);
+
+    _vertexArray;
+
+    bindGL();
+    glBufferData(
+        GL_ARRAY_BUFFER,
+        _vertexArray.size() * sizeof(float),
+        _vertexArray.data(),
+        GL_DYNAMIC_DRAW
+    );
+
+    //update vertex attributes
+    glVertexAttribPointer(_locVertex, 3, GL_FLOAT, GL_FALSE, 3 * sizeof(float), nullptr);
+
+    unbindGL();
+}
+
+void RenderableNodeLine::render(const RenderData& data, RendererTasks&) {
+
+    updateVertexData();
+    
+    _program->activate();
+
+    glm::dmat4 anchorTranslation(1.0);
+    // Update anchor node information, used to counter precision problems
+    if (global::navigationHandler.orbitalNavigator().anchorNode()) {
+        anchorTranslation = glm::translate(glm::dmat4(1.0), global::navigationHandler.orbitalNavigator().anchorNode()->worldPosition());
+    }
+
+    const glm::dmat4 modelTransform =
+        glm::translate(glm::dmat4(1.0), data.modelTransform.translation) *
+        glm::dmat4(data.modelTransform.rotation) *
+        glm::scale(glm::dmat4(1.0), glm::dvec3(data.modelTransform.scale));
+
+    const glm::dmat4 modelViewTransform = data.camera.combinedViewMatrix() *
+        modelTransform * anchorTranslation;
+
+    _program->setUniform("modelViewTransform", glm::mat4(modelViewTransform));
+    _program->setUniform("projectionTransform", data.camera.projectionMatrix());
+    _program->setUniform("color", glm::vec4(_lineColor.value(), _opacity));
+
+    // Save current state:
+    GLboolean isBlendEnabled = glIsEnabledi(GL_BLEND, 0);
+    GLboolean isLineSmoothEnabled = glIsEnabled(GL_LINE_SMOOTH);
+    GLfloat currentLineWidth;
+    glGetFloatv(GL_LINE_WIDTH, &currentLineWidth);
+
+    GLenum blendEquationRGB, blendEquationAlpha, blendDestAlpha,
+        blendDestRGB, blendSrcAlpha, blendSrcRGB;
+    glGetIntegerv(GL_BLEND_EQUATION_RGB, &blendEquationRGB);
+    glGetIntegerv(GL_BLEND_EQUATION_ALPHA, &blendEquationAlpha);
+    glGetIntegerv(GL_BLEND_DST_ALPHA, &blendDestAlpha);
+    glGetIntegerv(GL_BLEND_DST_RGB, &blendDestRGB);
+    glGetIntegerv(GL_BLEND_SRC_ALPHA, &blendSrcAlpha);
+    glGetIntegerv(GL_BLEND_SRC_RGB, &blendSrcRGB);
+
+    // Change GL state:
+    glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
+    glEnablei(GL_BLEND, 0);
+    glEnable(GL_LINE_SMOOTH);
+    glLineWidth(_lineWidth);
+
+    // Bind and draw
+    bindGL();
+    glDrawArrays(GL_LINES, 0, 2);
+    
+    // Restore GL State
+    unbindGL();
+    _program->deactivate();
+    glLineWidth(currentLineWidth);
+    glBlendEquationSeparate(blendEquationRGB, blendEquationAlpha);
+    glBlendFuncSeparate(blendSrcRGB, blendDestRGB, blendSrcAlpha, blendDestAlpha);
+    if (!isBlendEnabled) {
+        glDisablei(GL_BLEND, 0);
+    }
+    if (!isLineSmoothEnabled) {
+        glDisable(GL_LINE_SMOOTH);
+    }
+}
+
+
+/*  Returns a position that is relative to the current
```

Comment:
prefer `//` style of comments

Reasoning:  If only `//` stlye of comments are used in a file, it's possible to comment out large parts of it by using `/*  */` temporarily.  If `/*` is used instead, that doesn't work as:
```
/*
  commented out
  /*
    still commented out
  */
   *not* commented out
*/   Compile error
```